src/test/java/unit/PassthroughApps.java

```diff
@@ -0,0 +1,206 @@
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for the `terra app` commands and the pass-through apps: `terra gcloud`, `terra gsutil`,
+ * `terra bq`, `terra nextflow`.
+ */
+@Tag("unit")
+public class PassthroughApps extends SingleWorkspaceUnit {
+  @Test
+  @DisplayName("app list returns all pass-through apps")
+  void appList() throws IOException {
+    // `terra app list --format=json`
+    List<String> appList =
+        TestCommand.runAndParseCommandExpectSuccess(new TypeReference<>() {}, "app", "list");
+
+    // check that all pass-through apps are returned
+    assertTrue(appList.containsAll(Arrays.asList("gcloud", "gsutil", "bq", "nextflow")));
+  }
+
+  @Test
+  @DisplayName("env vars include workspace cloud project and pet SA key file")
+  @SuppressFBWarnings(
+      value = "NP_NULL_ON_SOME_PATH",
+      justification =
+          "Pet SA key file in the Context must be populated if the earlier login() call succeeded.")
+  void workspaceEnvVars() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    UFWorkspace workspace =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFWorkspace.class, "workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra app execute echo \$GOOGLE_APPLICATION_CREDENTIALS`
+    TestCommand.Result cmd =
+        TestCommand.runCommand("app", "execute", "echo", "$GOOGLE_APPLICATION_CREDENTIALS");
+
+    // check that GOOGLE_APPLICATION_CREDENTIALS = path to pet SA key file
+    assertTrue(
+        cmd.stdOut.contains(Context.getPetSaKeyFile().getFileName().toString()),
+        "GOOGLE_APPLICATION_CREDENTIALS set to pet SA key file");
+
+    // `terra app execute echo \$GOOGLE_APPLICATION_CREDENTIALS`
+    cmd = TestCommand.runCommand("app", "execute", "echo", "$GOOGLE_CLOUD_PROJECT");
+
+    // check that GOOGLE_CLOUD_PROJECT = workspace project
+    assertTrue(
+        cmd.stdOut.contains(workspace.googleProjectId),
+        "GOOGLE_CLOUD_PROJECT set to workspace project");
+  }
+
+  @Test
+  @DisplayName("env vars include a resolved workspace resource")
+  void resourceEnvVars() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra resources create gcs-bucket --name=$name --bucket-name=$bucketName --format=json`
+    String name = "resourceEnvVars";
+    String bucketName = UUID.randomUUID().toString();
+    TestCommand.runCommandExpectSuccess(
+        "resources", "create", "gcs-bucket", "--name=" + name, "--bucket-name=" + bucketName);
+
+    // `terra app execute echo \$TERRA_$name`
+    TestCommand.Result cmd = TestCommand.runCommand("app", "execute", "echo", "$TERRA_" + name);
```

Comment:
These env vars are populated in the Docker container the CLI uses to launch apps. So the env vars are available to apps, but don't affect the process you're calling from. We can't actually modify the environment variables in the calling environment from Java -- we'd have to e.g. generate a bash script with all the env vars we want and then tell the user to run `source generatedscript.sh`. That seems kind of hacky, especially because we wouldn't be able to keep them up to date when e.g. a user adds a new resource. But it might be a useful thing to users who want to use tools they've got installed locally. Anyway, it's something we've discussed but without a specific use case, it hasn't made it to the top of the priority list.